### PR TITLE
Allow to pass `Capabilities` and `Extensions` when creating an `OpenMlsLeafNode`.

### PR DESCRIPTION
**Pull request by duesee**
This (rather) resolves openmls/annotations#33 and resolves an `xxx:`s on the way.

### CODE REVIEW

File: openmls/src/treesync/node/leaf_node.rs [hunk 1]
```diff
@@ -718,12 +747,15 @@ impl OpenMlsLeafNode {
         leaf_node_source: LeafNodeSource,
         backend: &impl OpenMlsCryptoProvider,
         credential_bundle: &CredentialBundle,
+        capabilities: Capabilities,
```
Comment (franziskuskiefer) [inline openmls/src/treesync/node/leaf_node.rs:750]: Looks like comments need to be updated here on the function (and maybe elswhere).

File: openmls/src/key_packages/mod.rs [hunk 2]
```diff
@@ -267,6 +267,7 @@ impl KeyPackage {
             config,
             credential, // FIXME
             LeafNodeSource::KeyPackage(Lifetime::default()),
+            Capabilities::default(),
```
Comment (franziskuskiefer) [inline openmls/src/key_packages/mod.rs:270]: So we still don't set them. Let's do a follow up to allow actually passing in values.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks. Some comment needs updating. Otherwise this is good to go.

I see at least one follow up already that we should either file or tackle right after.

File: openmls/src/key_packages/mod.rs (hunk 2 above)
Comment (duesee) [reply openmls/src/key_packages/mod.rs:270]: Yeah. I tried to not touch key packages here. Will do a follow-up :+1:

Comment (duesee) [reply openmls/src/key_packages/mod.rs:270]: https://github.com/openmls/openmls/issues/1195

**Review by franziskuskiefer — DISMISSED** [review]
Thanks

**Comment by franziskuskiefer** [comment]
looks like `rust fmt` isn't happy

**Comment by duesee** [comment]
Urgs... now it should :+1:

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**